src/app/components/activity/feed/view.ts

```diff
@@ -149,6 +151,45 @@ export class ActivityFeedView {
 		}
 	}
 
+	processPinned(input: EventItem) {
+		const inputIsGame = input.action.game instanceof Game;
+
+		for (const item of this.state.items) {
+			if (
+				item.feedItem instanceof EventItem &&
+				item.feedItem.action instanceof FiresidePost &&
+				input.action instanceof FiresidePost
+			) {
+				// Same post that got pinned
+				if (item.feedItem.action.id === input.action.id) {
+					return;
+				}
+
+				const itemIsGame = item.feedItem.action.game instanceof Game;
+				// Belongs to different game.
+				if (itemIsGame && inputIsGame) {
+					if (item.feedItem.action.game.id !== input.action.game.id) {
+						return;
+					}
+				}
+
+				// Not the same feed types
+				if (itemIsGame !== inputIsGame) {
+					return;
+				}
+
+				// Not a user post from the same user
+				if (!itemIsGame && !inputIsGame) {
+					if (input.action.user.id !== item.feedItem.action.user.id) {
+						return;
+					}
+				}
+
+				item.feedItem.action.is_pinned = false;
```

Comment:
I don't like this - the post is pinned, just not in the feed so setting the is_pinned to false might have unknown side effects if we use the same instance of the post in other places besides the feed. this should probably be some state on the feed item or a getter on the feed service.